code/datums/forge.dm

```diff
@@ -0,0 +1,354 @@
+				F.force_wielded = F.force_wielded * 1.3
+				F.throwforce = F.throwforce * 1.3
+			if(FORGED_BULLET_CASING)
+				var/obj/projectile/bullet/forged/F = I
+				F.damage = F.damage * 1.3
+				F.armour_penetration += 10
+
+
+/datum/special_trait/sharp
+	name = "Sharp"
+	desc = "Makes sharp weapons sharper, increases armour penetration and shifts reagent transfer more towards direct injection as opposed to touch"
+
+
+/datum/special_trait/sharp/on_apply(obj/item/I, type)
+	if(type && I)
+		switch(type)
+			if(FORGED_MELEE_SINGLEHANDED)
+				var/obj/item/forged/F = I
+				F.armour_penetration += 10
+				F.stabby += TRANSFER_SHARP_BONUS
+			if(FORGED_MELEE_TWOHANDED)
+				var/obj/item/twohanded/forged/F = I
+				F.armour_penetration += 10
+				F.stabby += TRANSFER_SHARP_BONUS
+			if(FORGED_BULLET_CASING)
+				var/obj/projectile/bullet/forged/F = I
+				F.dismemberment += 20
+
+
+/datum/special_trait/radioactive
+	name = "Radioactive"
+	desc = "Continously irradiates the surrounding area, quite dangerous to the user!"
+
+
+/datum/special_trait/radioactive/on_apply(obj/item/I, type)
+	if(type && I)
+		switch(type)
+			if(FORGED_MELEE_SINGLEHANDED)
+				var/obj/item/forged/F = I
+				F.radioactive = TRUE
+				START_PROCESSING(SSobj, F)
+			if(FORGED_MELEE_TWOHANDED)
+				var/obj/item/twohanded/forged/F = I
+				F.radioactive = TRUE
+				START_PROCESSING(SSobj, F)
+			if(FORGED_BULLET_CASING)
+				var/obj/projectile/bullet/forged/F = I
+				F.radioactive = TRUE
+
+
+/datum/special_trait/ultradense
+	name = "Ultradense"
+	desc = "Found in exceptionally dense materials, allows bullets to penetrate and adds knockback to all weapons"
+
+
+/datum/special_trait/ultradense/on_apply(obj/item/I, type)
+	if(type && I)
+		switch(type)
+			if(FORGED_BULLET_CASING)
+				var/obj/projectile/bullet/forged/F = I
+				if(F.damage >= 20)
+					F.projectile_piercing = PASSMOB
+					F.projectile_phasing = (ALL & (~PASSMOB))
+
+
+/datum/special_trait/ultradense/on_hit(atom/target, mob/user, obj/item/I, type)
+	if(I && target && type)
+		if(type == FORGED_BULLET_CASING)
+			var/obj/projectile/bullet/forged/F = I
+			if(isliving(target))
+				var/mob/living/M = target
+				M.throw_at(get_edge_target_turf(M, F.dir),1 ,5)
+		else//we assume it's melee
+			if(isliving(target))
+				var/mob/living/M = target
+				M.throw_at(get_edge_target_turf(M, user.dir),1 ,5)
+
+
+/datum/special_trait/magnetic
+	name = "Magnetic"
+	desc = "High chance of throwing all metal items to the object's location when activated"
+	effectiveness = 50
+
+
+/datum/special_trait/magnetic/on_hit(atom/target, mob/user, obj/item/I, type)
+	if(I && type)
+		if(type == FORGED_BULLET_CASING)
+			for(var/obj/O in orange(I, 5))
+				if(O && !O.anchored && O.flags_1 & CONDUCT_1)
+					O.throw_at(I, 4, 3)
+		else if(user)
+			for(var/obj/O in orange(user, 5))
+				if(O && !O.anchored && O.flags_1 & CONDUCT_1)
+					O.throw_at(user, 4, 3)
+
+/datum/special_trait/reflective
```

Comment:
probably could use a better name for this one

Reply:
Got a better idea?

Reply:
i dunno deflection? its just a bizarre name if its upping block chance

Reply:
Changed.